var/spack/repos/builtin/packages/py-mvdtool/package.py

```diff
@@ -15,15 +15,30 @@ class PyMvdtool(PythonPackage):
     git      = "https://github.com/BlueBrain/MVDTool.git"
 
     version('develop', branch='master', submodules=True, clean=False)
+    version('2.3.0', tag='v2.3.0', submodules=True, clean=False)
+    version('2.2.1', tag='v2.2.1', submodules=True, clean=False)
     version('2.2.0', tag='v2.2.0', submodules=True, clean=False)
     version('2.1.0', tag='v2.1.0', submodules=True, clean=False)
     version('2.0.0', tag='v2.0.0', submodules=True, clean=False)
 
+    variant('mpi', default=True, description='Build with support for MPI')
+
     depends_on('py-setuptools', type='build')
     depends_on('py-setuptools-scm', type='build')
 
     depends_on('cmake@3.2:', type='build')
     depends_on('py-numpy', type='run')
-    depends_on('hdf5~mpi', type=('build', 'run'))
-    depends_on('highfive~mpi', type='build')
-    depends_on('libsonata~mpi', type=('build', 'run'), when='@2.1:')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('hdf5+mpi', type=('build', 'run'), when="+mpi")
+    depends_on('hdf5~mpi', type=('build', 'run'), when="~mpi")
+    depends_on('libsonata~mpi', type=('build', 'run'), when='~mpi@2.1:')
+    depends_on('libsonata+mpi', type=('build', 'run'), when='+mpi@2.1:')
+    depends_on('highfive~mpi', type='build', when='~mpi')
+    depends_on('highfive+mpi', type='build', when='+mpi')
+
+    @when('+mpi')
+    @run_before('build')
+    def configure(self):
```

Comment:
Not sure if these decorators stack the right way? Seems like that's triggering the failure. Maybe swapping them would help? If not, I think the `mpi` test should be moved inside the function.

Reply:
Indeed we cannot stack them :( The other way around gives MethodNotFound